File: 2015WorksApplications/internship/intern/Worksap/workspace/China_Intern/China_Intern_Java/src/jp/co/worksap/intern/entities/hotel/HotelMstDTOReader.java
package jp.co.worksap.intern.entities.hotel;

import java.io.IOException;

import jp.co.worksap.intern.constants.Constants;
import jp.co.worksap.intern.entities.AbstractDTOReader;

public class HotelMstDTOReader extends AbstractDTOReader<HotelMstDTO> {
	private static final int COLUMN_INDEX_HOTEL_ID = 0;
	private static final int COLUMN_INDEX_HOTEL_NAME = 1;
	private static final int COLUMN_INDEX_ADDRESS = 2;
	private static final int COLUMN_INDEX_TEL = 3;
	private static final int COLUMN_INDEX_STARS = 4;

	private String fileAddress = Constants.DEFAULT_CSV_FOLDER + "HOTEL_MST.csv";

	/**
	 * use default file address
	 * 
	 * @throws IOException
	 */
	public HotelMstDTOReader() throws IOException {
		super(HotelMstDTOReader.class.getName());
		super.init();
	}

	/**
	 * use customize file address
	 * 
	 * @param fileAddress
	 * @throws IOException
	 */
	public HotelMstDTOReader(final String fileAddress) throws IOException {
		super(HotelMstDTOReader.class.getName());
		this.fileAddress = fileAddress;
		super.init();
	}

	@Override
	protected String getFileArress() {
		return fileAddress;
	}

	@Override
	protected HotelMstDTO convertArrayToDTO(String[] value) throws IOException {
		Long hotelId = Long.valueOf(value[COLUMN_INDEX_HOTEL_ID]);
		String hotelName = value[COLUMN_INDEX_HOTEL_NAME];
		String address = value[COLUMN_INDEX_ADDRESS];
		String tel = value[COLUMN_INDEX_TEL];
		Integer stars = Integer.valueOf(value[COLUMN_INDEX_STARS]);

		HotelMstDTO dto = new HotelMstDTO(hotelId,hotelName,address,tel,stars);
		return dto;
	}
}

File: 2015WorksApplications/internship/b032/Final_submission/VS2012Solution/src/entities/hotel/hotel_mst_dto_reader.cpp
#include "hotel_mst_dto_reader.h"
#include "hotel_mst_dto.h"
#include "constants/constants.h"
#include "entities/icsv_master_dto.h"
#include <string>
#include <vector>
#include <cstdlib>
using namespace std;

HotelMstDTOReader::HotelMstDTOReader() {
	this->file_address = kDefaultCsvFolder + "HOTEL_MST.csv";
	AbstractDTOReader::init();
}

HotelMstDTOReader::HotelMstDTOReader(string &file_address) {
	this->file_address = file_address;
	AbstractDTOReader::init();
}

string HotelMstDTOReader::get_file_address() {
	return this->file_address;
}

ICsvMasterDTO *HotelMstDTOReader::convert_array_to_dto(vector<string> &value) {
	long hotel_id = atol(value[COLUMN_INDEX_HOTEL_ID].c_str());
	string hotel_name = value[COLUMN_INDEX_HOTEL_NAME];
	string address = value[COLUMN_INDEX_ADDRESS];
	string tel = value[COLUMN_INDEX_TEL];
	ICsvMasterDTO *obj = new HotelMstDTO(hotel_id, hotel_name, address, tel, 0 /* stars */);
	return obj;
}

HotelMstDTOReader::~HotelMstDTOReader() {

}

